fecgov/openFEC
# 5522-add cmte_tp, filing_form filters add dep notice

## Summary 

- Resolves #5522 

Adds cmte_tp and filing_form filters to the debts endpoint. Adds deprecation notice to candidate_id filter. Candidate_id is <1% non-null. Removes mostly null fields from debts endpoint [to support the request to remove null fields from debts export. ](https://github.com/fecgov/openFEC/issues/5513)

### Required reviewers

2 devs

## Impacted areas of the application

-  Debts endpoint

## Screenshots

![Screen Shot 2023-08-14 at 11 45 46 AM](https://github.com/fecgov/openFEC/assets/66386084/e744795a-739e-4127-b88e-d837ae3a864e)

Mostly null fields %'s
![Screen Shot 2023-08-14 at 11 46 09 AM](https://github.com/fecgov/openFEC/assets/66386084/d2ebb046-63cb-4688-aef9-75de33a47347)


## Related PRs


## How to test
**Because I am removing fields, we need to include the migration file with the python changes. Do not deploy this PR to any space**
- Pull this PR
- in [webservices/common/models/itemized.py] change line 492 from:
__tablename__ = 'ofec_sched_d_mv'
to 
__tablename__ = 'ofec_sched_d_mv_cl_temp'

- make sure you're connected to dev
- flask run
-  http://127.0.0.1:5000/v1/schedules/schedule_d/?committee_type=H&page=1&per_page=20&sort=-coverage_end_date&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false
-  http://127.0.0.1:5000/v1/schedules/schedule_d/?filing_form=F3X&page=1&per_page=20&sort=-coverage_end_date&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false
- http://127.0.0.1:5000/v1/schedules/schedule_d/?&page=1&per_page=20&sort=-coverage_end_date&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false
- http://127.0.0.1:5000/v1/schedules/schedule_d/?line_number=9&committee_type=P&page=1&per_page=20&sort=-coverage_end_date&sort_hide_null=false&sort_null_only=false&sort_nulls_last=false

I will delete my temp MV after review.
